applications/minotari_merge_mining_proxy/src/block_template_protocol.rs

```diff
@@ -375,26 +401,26 @@ pub fn calculate_aux_chain_merkle_root(hashes: Vec<monero::Hash>) -> Result<(mon
 fn add_monero_data(
     tari_block_result: grpc::GetNewBlockResult,
     monero_mining_data: MoneroMiningData,
-    template_data: NewBlockTemplateData,
+    miner_data: MinerData,
 ) -> Result<FinalBlockTemplateData, MmProxyError> {
     let merge_mining_hash = FixedHash::try_from(tari_block_result.merge_mining_hash.clone())
         .map_err(|e| MmProxyError::ConversionError(e.to_string()))?;
 
     let aux_chain_hashes = vec![monero::Hash::from_slice(merge_mining_hash.as_slice())];
-    let tari_difficulty = template_data.miner_data.target_difficulty;
+    let tari_difficulty = miner_data.target_difficulty;
```

Comment:
If the block comes from p2pool, target difficulty will be the original one. I can change the returning miner_data as well if needed.